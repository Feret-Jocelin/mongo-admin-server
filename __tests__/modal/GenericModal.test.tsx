/* eslint-disable no-undef */
import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { NextIntlClientProvider } from 'next-intl';
import GenericModal from '@/app/[locale]/components/modal/generic/GenericModal';

describe('Generic Modal', () => {
  const locale = 'fr';
  const message = require(`../../shared/locales/${locale}.json`)
  it("renders", () => {
    const handleClose = jest.fn();
    const handleValidate= jest.fn();
    const { container } = render(
      <NextIntlClientProvider locale={locale} messages={message}>
        <GenericModal 
          open={true}
          title="Title Generic Modal"
          onClose={handleClose}
          onValidate={handleValidate} 
        >
        </GenericModal>
      </NextIntlClientProvider>
    )
    expect(screen.findByText('Title Generic Modal')).toBeInTheDocument
    expect(container).toMatchSnapshot();
  })
})