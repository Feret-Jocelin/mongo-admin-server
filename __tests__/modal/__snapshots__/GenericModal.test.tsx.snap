// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generic Modal renders 1`] = `
<div>
  <dialog
    class="modal"
  >
    <section
      class="modal__header"
    >
      <div
        class="modal__header--icon"
      >
        <div
          class="undefined pointer-cursor"
        />
      </div>
      <h2>
        Title Generic Modal
      </h2>
    </section>
    <section
      class="modal__container"
    />
    <section
      class="modal__footer"
    >
      <button
        class="button full outline center"
        type="button"
      >
        Annuler
      </button>
      <button
        class="button full center"
        type="button"
      >
        Confirmer
      </button>
    </section>
  </dialog>
</div>
`;
